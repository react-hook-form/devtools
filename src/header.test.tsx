/* eslint-disable @typescript-eslint/no-empty-function */
import * as React from 'react';
import Header from './header';
import { render } from '@testing-library/react';

describe('header', () => {
  it('should render correct color when form pass', () => {
    const methods = render(
      <Header
        setVisible={() => {}}
        control={
          {
            formState: { isValid: true },
            readFormStateRef: { current: {} },
          } as any
        }
      />,
    );
    expect(methods.asFragment()).toMatchSnapshot();
  });

  it('should render correct when form failed', () => {
    const methods = render(
      <Header
        setVisible={() => {}}
        control={
          {
            formState: { isValid: false },
            readFormStateRef: { current: {} },
          } as any
        }
      />,
    );
    expect(methods.asFragment()).toMatchSnapshot();
  });
});
