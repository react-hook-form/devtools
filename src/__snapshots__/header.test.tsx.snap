// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render correct color when form pass 1`] = `
<DocumentFragment>
  <header
    style="display: flex; align-items: center; padding-left: 10px;"
  >
    <p
      style="margin: 0px; padding: 0px; font-weight: 600;"
    >
      <span
        style="transition: 0.5s all; color: rgb(27, 218, 43);"
      >
        ■
      </span>
       RHF DevTools
    </p>
    <button
      class="css-7zizgu"
      title="Close dev panel"
    >
      ✕
    </button>
  </header>
</DocumentFragment>
`;

exports[`header should render correct when form failed 1`] = `
<DocumentFragment>
  <header
    style="display: flex; align-items: center; padding-left: 10px;"
  >
    <p
      style="margin: 0px; padding: 0px; font-weight: 600;"
    >
      <span
        style="transition: 0.5s all; color: rgb(236, 89, 144);"
      >
        ■
      </span>
       RHF DevTools
    </p>
    <button
      class="css-7zizgu"
      title="Close dev panel"
    >
      ✕
    </button>
  </header>
</DocumentFragment>
`;
